shared/src/bad_token/trace_call.rs

```diff
@@ -93,32 +93,17 @@ impl BadTokenDetecting for TraceCallDetector {
 
 impl TraceCallDetector {
     pub async fn detect_impl(&self, token: H160) -> Result<TokenQuality> {
-        let instance = ERC20::at(&self.web3, token);
-        let decimals = match instance.decimals().call().await {
-            Ok(decimals) => decimals,
-            Err(err) => {
-                return match EthcontractErrorType::classify(&err) {
-                    EthcontractErrorType::Node => Err(err.into()),
-                    EthcontractErrorType::Contract => {
-                        Ok(TokenQuality::bad("failed to get decimals"))
-                    }
-                }
-            }
-        };
-        let amount = match U256::from(10).checked_pow(decimals.into()) {
-            Some(amount) => {
-                tracing::debug!("testing token {:?} with amount {}", token, amount);
-                amount
-            }
-            None => return Ok(TokenQuality::bad("decimals too large")),
-        };
-
-        let take_from = match self.find_largest_pool_owning_token(token).await? {
-            Some((address, balance)) if balance > amount => {
```

Comment:
If I understand `find_largest_pool_owning_token` correctly, `balance` can be `0`, so we still need at least some check here.

Reply:
I've changed it to check for 0 balance. So it either returns Some with an address that has more than 0 or None.